### PR TITLE
Moved the serialisation logic in nodeapi.internal.serialization into its own module: serialization

### PR DESCRIPTION
**Pull request by shamsasari**
node-api now depends on this module and upcoming changes will use this as well rather than having to depend on node-api.

EnumEvolveTests.deserializeWithRename and EnumEvolveTests.multiOperations are temporarily ignored since their test resources can't be regenerated due to bugs.

### CODE REVIEW

File: serialization/src/main/kotlin/net/corda/serialization/internal/amqp/custom/X509CRLSerializer.kt [hunk 1]
```diff
@@ -17,16 +17,12 @@ object X509CRLSerializer : CustomSerializer.Implements<X509CRL>(X509CRL::class.j
             emptyList()
     )))
 
-    override fun writeDescribedObject(obj: X509CRL, data: Data, type: Type, output: SerializationOutput,
-                                      context: SerializationContext
-    ) {
+    override fun writeDescribedObject(obj: X509CRL, data: Data, type: Type, output: SerializationOutput, context: SerializationContext) {
```
Comment (fenryka) [inline serialization/src/main/kotlin/net/corda/serialization/internal/amqp/custom/X509CRLSerializer.kt]: line too long, don't undo wrapping

File: serialization/src/main/kotlin/net/corda/serialization/internal/amqp/custom/X509CRLSerializer.kt [hunk 2]
```diff
@@ -17,16 +17,12 @@ object X509CRLSerializer : CustomSerializer.Implements<X509CRL>(X509CRL::class.j
...
         output.writeObject(obj.encoded, data, clazz, context)
     }
 
-    override fun readObject(obj: Any, schemas: SerializationSchemas, input: DeserializationInput,
-                            context: SerializationContext
-    ): X509CRL {
+    override fun readObject(obj: Any, schemas: SerializationSchemas, input: DeserializationInput, context: SerializationContext): X509CRL {
```
Comment (fenryka) [inline serialization/src/main/kotlin/net/corda/serialization/internal/amqp/custom/X509CRLSerializer.kt]: line too long, don't undo wrapping

File: serialization/src/main/kotlin/net/corda/serialization/internal/amqp/custom/X509CertificateSerializer.kt [hunk 3]
```diff
@@ -17,16 +17,12 @@ object X509CertificateSerializer : CustomSerializer.Implements<X509Certificate>(
...
-                                      context: SerializationContext
-    ) {
+    override fun writeDescribedObject(obj: X509Certificate, data: Data, type: Type, output: SerializationOutput, context: SerializationContext) {
         output.writeObject(obj.encoded, data, clazz, context)
     }
 
-    override fun readObject(obj: Any, schemas: SerializationSchemas, input: DeserializationInput,
-                            context: SerializationContext
-    ): X509Certificate {
+    override fun readObject(obj: Any, schemas: SerializationSchemas, input: DeserializationInput, context: SerializationContext): X509Certificate {
```
Comment (fenryka) [inline serialization/src/main/kotlin/net/corda/serialization/internal/amqp/custom/X509CertificateSerializer.kt]: line too long, don't undo wrapping

File: serialization/src/main/kotlin/net/corda/serialization/internal/amqp/custom/X509CertificateSerializer.kt [hunk 4]
```diff
@@ -17,16 +17,12 @@ object X509CertificateSerializer : CustomSerializer.Implements<X509Certificate>(
             emptyList()
     )))
 
-    override fun writeDescribedObject(obj: X509Certificate, data: Data, type: Type, output: SerializationOutput,
-                                      context: SerializationContext
-    ) {
+    override fun writeDescribedObject(obj: X509Certificate, data: Data, type: Type, output: SerializationOutput, context: SerializationContext) {
```
Comment (fenryka) [inline serialization/src/main/kotlin/net/corda/serialization/internal/amqp/custom/X509CertificateSerializer.kt]: line too long, don't undo wrapping

File: serialization/src/test/kotlin/net/corda/serialization/internal/amqp/EnumEvolveTests.kt [hunk 5]
```diff
@@ -271,6 +275,9 @@ class EnumEvolveTests {
     // Finally, the original version of teh class that we're going to be testing with
     enum class MultiOperations { A, B, C }
 
+    // TODO See Ignore message below
```
Comment (fenryka) [inline serialization/src/test/kotlin/net/corda/serialization/internal/amqp/EnumEvolveTests.kt]: Log Jira

File: serialization/src/test/kotlin/net/corda/serialization/internal/amqp/EnumEvolveTests.kt [hunk 6]
```diff
@@ -149,6 +150,9 @@ class EnumEvolveTests {
     // Finally, the version we're using to test with
     enum class DeserializeWithRename { A, B, C }
 
+    // TODO See Ignore message below
+    @Ignore("The resources for this test need regenerating however that's not possible as the 3rd change set returns a " +
```
Comment (fenryka) [inline serialization/src/test/kotlin/net/corda/serialization/internal/amqp/EnumEvolveTests.kt]: Log Jira

File: serialization/src/test/kotlin/net/corda/serialization/internal/carpenter/EnumClassTests.kt [hunk 7]
```diff
@@ -1,6 +1,6 @@
-package net.corda.nodeapi.internal.serialization.carpenter
+package net.corda.serialization.internal.carpenter
 
-import net.corda.nodeapi.internal.serialization.AllWhitelist
+import net.corda.serialization.internal.kryo.AllWhitelist
```
Comment (fenryka) [inline serialization/src/test/kotlin/net/corda/serialization/internal/carpenter/EnumClassTests.kt]: I think I noticed this above, but if AMQP stuff needs AllWhitelist that should move into a common module

**Comment by fenryka** [comment]
As the horrible list of conflicts above indicates... with AMQP RPC now in the tree... this might need a hell of a rebase :)

**Comment by shamsasari** [comment]
It's needed by one of the tests, `AttachmentsClassLoaderTests` I believe.

**Pull request merged**